frame/executive/src/lib.rs

```diff
@@ -589,16 +670,22 @@ where
 		//
 		// The entire block should be discarded if an inherent fails to apply. Otherwise
 		// it may open an attack vector.
-		if r.is_err() && dispatch_info.class == DispatchClass::Mandatory {
+		let mandatory = dispatch_info.class == DispatchClass::Mandatory;
+		if r.is_err() && mandatory {
 			return Err(InvalidTransaction::BadMandatory.into())
 		}
+		if mode == ExtrinsicInclusionMode::OnlyInherents && !mandatory {
+			// Note: The block builder should never try to do this.
```

Comment:
Why? imagine we are running MBMs, and there is some non-mandatory inherent (of which we currently don't have an example, but it could exist). 

It seems like `ExtrinsicInclusionMode::OnlyInherents` is actually `ExtrinsicInclusionMode::OnlyMandatoryInherents`? 🙈

Reply:
There could still be optional inherents, the blockbuilder would just omit them.  

Mandatory has two meanings that can get mixed up here: Mandatory in the sense of being infallible and Mandatory in the sense that it needs to execute in every block. We can still have optional inherents that are non-mandatory in the second sense by letting the blockbuilder omit them.  
But having an inherents that can fail is not quite useful i think. LMK if you meant something else :smile: 

Not sure about the `OnlyMandatoryInherents`, maybe but since there are no non-mandatory it is not adding any information, or?